src/utils/getAddressLabel.ts

```diff
@@ -0,0 +1,23 @@
+import type { GetEnsNameReturnType } from 'wagmi/actions';
+import { walletDigest } from './walletDigest';
+interface getAddressLabelProps {
+  isSuccess: boolean;
+  ensName?: GetEnsNameReturnType;
+  address?: string;
+}
+
+export const getAddressLabel = ({
+  isSuccess,
+  ensName,
+  address,
+}: getAddressLabelProps): string => {
+  if (isSuccess && ensName) {
+    return String(ensName).length > 20
+      ? `${String(ensName).slice(0, 13)}...eth`
+      : ensName;
+  } else if (address) {
+    return walletDigest(address);
+  } else {
+    return '0x0000...0000';
```

Comment:
please use walletDigest function for this, seems to be missing 1 char

Reply:
Good idea! Please give it another look as I refactored that part